synthesizer/src/vm/mod.rs

```diff
@@ -405,8 +405,8 @@ impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
                 // Unpause the atomic writes, executing the ones queued from block insertion and finalization.
                 #[cfg(feature = "rocks")]
                 self.block_store().unpause_atomic_writes::<false>()?;
-                // If the block advances to `ConsensusVersion::V4`, clear the partial verification cache.
-                if N::CONSENSUS_HEIGHT(ConsensusVersion::V4).unwrap_or_default() == block.height() {
+                // If the block advances to a new consensus version, clear the partial verification cache.
+                if N::CONSENSUS_VERSION_HEIGHTS.iter().any(|(_, height)| height == &block.height()) {
```

Comment:
Performance nit: we can reverse iter and break the loop if the block height is below CONSENSUS_VERSION_HEIGHTS

Reply:
Its a nit now but will be essential once we have a hundred versions